lib/ansible/modules/storage/glusterfs/gluster_heal_facts.py

```diff
@@ -0,0 +1,166 @@
+        if rc != 0:
+            module.fail_json(msg='error running gluster (%s) command (rc=%d): %s' %
+                             (' '.join(args), rc, out or err), exception=traceback.format_exc())
+    except Exception as e:
+        module.fail_json(msg='error running gluster (%s) command: %s' % (' '.join(args),
+                                                                         to_native(e)), exception=traceback.format_exc())
+    return out
+
+
+def get_self_heal_status(name):
+    out = run_gluster(['volume', 'heal', name, 'info'])
+    raw_out = out.split("\n")
+    heal_info = []
+    # return files that still need healing.
+    for line in raw_out:
+        if 'Brick' in line:
+            br_dict = {}
+            br_dict['brick'] = line.strip().strip("Brick")
+        elif 'Status' in line:
+            br_dict['status'] = line.split(":")[1].strip()
+        elif 'Number' in line:
+            br_dict['no_of_entries'] = line.split(":")[1].strip()
+            if int(br_dict['no_of_entries']) != 0:
+                heal_info.append(br_dict)
+            else:
+                continue
+    return heal_info
+
+
+def get_rebalance_status(name):
+    out = run_gluster(['volume', 'rebalance', name, 'status'])
+    return out
+
+
+def main():
+    global module
+    argument_spec = dict(name=dict(type='str', required=True, aliases=['volume']),
+                         status_filter=dict(default='self-heal', choices=['self-heal', 'rebalance']),
+                         )
+    module = AnsibleModule(argument_spec)
+
+    global glusterbin
+    glusterbin = module.get_bin_path('gluster', True)
+
+    required_version = "3.2"
+    status_filter = module.params['status_filter']
+    volume_name = module.params['name']
+    heal_info = ''
+    rebalance_status = ''
+
+    # Verify if required GlusterFS version is installed
+    if is_invalid_gluster_version(module, required_version):
+        module.fail_json(msg="GlusterFS version > %s is required" %
+                         required_version)
+
+    try:
+        if status_filter == "self-heal":
+            heal_info = get_self_heal_status(volume_name)
+        elif status_filter == "rebalance":
+            rebalance_status = get_rebalance_status(volume_name)
+    except Exception:
+        module.fail_json(msg="Invalid heal status option.")
+
+    facts = {}
+    facts['glusterfs'] = {'volume': volume_name, 'status_filter': status_filter, 'heal_info': heal_info, 'rebalance': rebalance_status}
+
+    module.exit_json(ansible_facts=facts)
+
+
+def is_invalid_gluster_version(module, required_version):
+    cmd = module.get_bin_path('gluster', True) + ' --version'
+    result = module.run_command(cmd)
+    ver_line = result[1].split('\n')[0]
+    version = ver_line.split(' ')[1]
+    # If the installed version is less than 3.2, it is an invalid version
+    # return True
+    return LooseVersion(version) < LooseVersion(required_version)
+
```

Comment:
Can you move this function above main() as per ansible guildelines:

" 
Ansible follows C-style code flow where the caller functions/methods are towards the bottom of the file and the callee implementations are above them.
"

Reply:
Done.